### PR TITLE
feat: offer to install missing Lean using elan

### PR DESCRIPTION
**Pull request by semorrison**
Seems to work nicely on OS X, haven't had a chance to test elsewhere.

### CODE REVIEW

**Comment by semorrison** [comment]
I've just updated this so it overrides the path to Lean after attempting to install Lean via elan, since the $PATH won't yet be set correctly.

This seems to work quite well on OS X. I haven't tested it elsewhere, but perhaps we can consider merging anyway? I will attempt to test in virtual machines soon, but as it may just work, and shouldn't break anything, it seems reasonable to get people to test it in the wild. :-)

**Comment by semorrison** [comment]
I've just improved this so on Windows we look for git bash, and if found install elan inside a git bash shell.

**Comment by semorrison** [comment]
I've now tested this on Windows using both MSYS2, and git bash. Both now work nicely!

**Comment by gebner** [comment]
Looks good otherwise (didn't try it though).

File: src/server.ts [hunk 1]
```diff
@@ -55,6 +62,20 @@ export class Server extends leanclient.Server {
             // TODO(gabriel): unset LEAN_PATH environment variable
 
             this.executablePath = config.get('executablePath') || 'lean';
+            if(this.executablePath === 'lean') {
```
[the remark this answers is not in the document]
Comment (semorrison) [reply src/server.ts]: Done!

File: src/server.ts [hunk 2]
```diff
@@ -2,8 +2,13 @@ import * as leanclient from 'lean-client-js-node';
 import { CommandResponse, CompleteResponse, Event, FileRoi, Message,
      ProcessConnection, ProcessTransport, RoiRequest, Task } from 'lean-client-js-node';
 import * as semver from 'semver';
-import { OutputChannel, window, workspace } from 'vscode';
+import { commands, OutputChannel, window, workspace } from 'vscode';
 import { LowPassFilter } from './util';
+import { fstat } from 'fs';
+const path = require('path');
+const osHomedir = require('os-homedir');
```
[the remark this answers is not in the document]
Comment (semorrison) [reply src/server.ts]: Done!

File: src/server.ts [hunk 3]
```diff
@@ -2,8 +2,13 @@ import * as leanclient from 'lean-client-js-node';
 import { CommandResponse, CompleteResponse, Event, FileRoi, Message,
      ProcessConnection, ProcessTransport, RoiRequest, Task } from 'lean-client-js-node';
 import * as semver from 'semver';
-import { OutputChannel, window, workspace } from 'vscode';
+import { commands, OutputChannel, window, workspace } from 'vscode';
 import { LowPassFilter } from './util';
+import { fstat } from 'fs';
```
[the remark this answers is not in the document]
Comment (semorrison) [reply src/server.ts]: Removed!

**Comment by semorrison** [comment]
Okay, I've just quickly retested (just on macOS), and everything still seems to work fine.

When this gets merged, I'll make some 'video walkthroughs' of the setup process, too.

**Pull request merged**